src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -4671,7 +4702,7 @@ internal Buffer(IEnumerable<TElement> source)
 
         internal TElement[] ToArray()
         {
-            if (count == 0) return new TElement[0]; // consider replacing with Array.Empty<TElement>()
+            if (count == 0) return Array.Empty<TElement>();
```

Comment:
For just about every path, if `count == 0` then `array` should already be `Array.Empty<TElement>()`, so we can just cut this line and have the next line cover it.